src/helpers/blood-pressure-data-providers/survey-blood-pressure-data-provider.ts

```diff
@@ -0,0 +1,81 @@
+import MyDataHelps, { Guid, SurveyAnswer, SurveyAnswersPage, SurveyAnswersQuery } from "@careevolution/mydatahelps-js";
+import { BloodPressureDataPoint } from "./query-blood-pressure";
+import { format, parseISO, startOfDay, toDate } from "date-fns";
+
+export interface BloodPressureDataParameters {
+    surveyName: string,
+    dateOfResultId? : string,
+    systolicResultId? : string,
+    diastolicResultId? : string
+}
+
+export default async function(props : BloodPressureDataParameters) : Promise<BloodPressureDataPoint[]> {
+    const resultFilter = [props.dateOfResultId ?? "", props.systolicResultId ?? "", props.diastolicResultId ?? "" ];
+    
+
+    const surveyAnswers = await getSurveyAnswers(props.surveyName, resultFilter);
+    var sortedAnswers = (surveyAnswers).sort((a, b) => {
+            if (parseISO(a.date) > parseISO(b.date)) { return -1; }
+            if (parseISO(a.date) < parseISO(b.date)) { return 1; }
+            return 0;
+    });
+
+    return groupSurveyAnswersByResults(sortedAnswers);
+
+
+    function groupSurveyAnswersByResults(answers : SurveyAnswer[]){
+        let bpDataPoints : BloodPressureDataPoint[] = [];
+
+        if (answers.length > 0){
+            let resultIds = [...new Set(answers.map(a => a.surveyResultID ))];
+            resultIds.forEach( (resultId) => {
+                var resultsForSubmission = answers.filter(a => a.surveyResultID == resultId);
+                var bpLogDateResults = resultsForSubmission.find( r => r.resultIdentifier == props.dateOfResultId);
+                var bpLogDate =bpLogDateResults?.answers[0];
+                var bpSystolicResults = resultsForSubmission.find( r => r.resultIdentifier == props.systolicResultId);
+                var bpSystolic = bpSystolicResults?.answers[0];
```

Comment:
Good to guard against answers being an empty array, empty string, or invalid (e.g. non-numeric for systolic / diastolic, non-date for date)

Reply:
e.g. probably ignore/skip the result if it's actually invalid